plugins/anchor-links/index.test.js

```diff
@@ -110,82 +268,195 @@ describe('anchor-links', () => {
           }),
         ].join('\n')
       )
+      expect(headings).toMatchInlineSnapshot(`
+      Array [
+        Object {
+          "aliases": undefined,
+          "level": 1,
+          "permalinkSlug": "hello-world",
+          "slug": "hello-world",
+          "title": "hEllO----world",
+        },
+        Object {
+          "aliases": undefined,
+          "level": 1,
+          "permalinkSlug": "hello-world-1",
+          "slug": "hello-world-1",
+          "title": "hello :&-- world",
+        },
+        Object {
+          "aliases": undefined,
+          "level": 1,
+          "permalinkSlug": "hello-world-foo",
+          "slug": "hello-world-foo",
+          "title": "hello world (foo)()",
+        },
+      ]
+      `)
     })
 
     test('generates an extra slug if the compatibilitySlug argument is provided', () => {
+      const headings = []
       expect(
-        execute('# hello world', { compatibilitySlug: (_) => 'foo' })
+        execute('# hello world', { compatibilitySlug: (_) => 'foo', headings })
       ).toMatch(
         expectedHeadingResult({
           slug: 'hello-world',
           text: 'hello world',
           compatSlugs: ['foo'],
         })
       )
+      expect(headings).toMatchInlineSnapshot(`
+      Array [
+        Object {
+          "aliases": undefined,
+          "level": 1,
+          "permalinkSlug": "foo",
+          "slug": "hello-world",
+          "title": "hello world",
+        },
+      ]
+      `)
     })
 
     test('does not render duplicate compatibility slugs', () => {
+      const headings = []
       expect(
-        execute('# hello world', { compatibilitySlug: (_) => 'hello-world' })
+        execute('# hello world', {
+          compatibilitySlug: (_) => 'hello-world',
+          headings,
+        })
       ).toMatch(
         expectedHeadingResult({
           slug: 'hello-world',
           text: 'hello world',
         })
       )
+      expect(headings).toMatchInlineSnapshot(`
+      Array [
+        Object {
+          "aliases": undefined,
+          "level": 1,
+          "permalinkSlug": "hello-world",
+          "slug": "hello-world",
+          "title": "hello world",
+        },
+      ]
+      `)
     })
 
     test('anchor aliases', () => {
-      expect(execute('# hello world ((#foo))')).toMatch(
+      let headings = []
+      expect(execute('# hello world ((#foo))', { headings })).toMatch(
         expectedHeadingResult({
           slug: 'hello-world',
           text: 'hello world',
           compatSlugs: ['foo'],
         })
       )
+      expect(headings).toMatchInlineSnapshot(`
```

Comment:
Nice (to all new usages of `toMatchInlineSnapshot`)! Big fan of inline snapshots 😄

Reply:
I saw you use it in https://github.com/hashicorp/remark-plugins/pull/37 so I thought I'd give it a try too! It seems a bit easier to read than looping over an array of expected results and expecting attributes to have a certain value. My only qualm is that some of these take up a lot of vertical space, but I'm sure there are solutions for that!

Reply:
There is also [`toMatchSnapshot`](https://jestjs.io/docs/snapshot-testing), which puts the snapshot in a separate file (maybe you've used that before?).

I'm less of a fan of that as I find it very valuable to see the expected result right next to the test case.